### PR TITLE
Final "Build Image in Azure"

### LINKED ISSUE #286: Implement `Build from project remotely using Azure` command
This is a choice when creating a container app: 

image

### PR DESCRIPTION
**Pull request by motm32**
Fixes #286.

### CODE REVIEW

File: src/commands/deploy/buildImageInAzure/BuildImageStep.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    public priority: number = 225;
+
+    public async execute(context: IBuildImageInAzureContext): Promise<void> {
+        context.registryDomain = acrDomain;
+
+        const run = await buildImageInAzure(context);
+        const outputImages = run?.outputImages;
+        context.telemetry.properties.outputImages = outputImages?.length?.toString();
+
+        if (outputImages) {
```
Comment (nturinski) [inline src/commands/deploy/buildImageInAzure/BuildImageStep.ts:21]: I think we should throw an error here if there's not any output image. The whole create scenario will break at the end anyway, so at least we could give them a more descriptive error?

**Review by nturinski — APPROVED** [review]

File: src/commands/deploy/IDeployBaseContext.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ export interface IDeployBaseContext extends ISubscriptionActionContext {
     targetContainer?: ContainerAppModel;
 
     imageSource?: ImageSourceValues;
+    buildType?: string;
```
Comment (alexweininger) [inline src/commands/deploy/IDeployBaseContext.ts:17]: This should probably be a stronger type than just `string`. Either an enum or string union.

Comment (alexweininger) [inline src/commands/deploy/IDeployBaseContext.ts:17]: Also, we should try to document these context properties where necessary. How are these properties used and what do they control?

File: src/commands/deploy/buildImageInAzure/BuildFromProjectListStep.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { OSPickStep } from "./OSPickStep";
+import { RootFolderStep } from "./RootFolderStep";
+import { RunStep } from "./RunStep";
+import { TarFileStep } from "./TarFileStep";
+import { UploadSourceCodeStep } from "./UploadSourceCodeStep";
+
+const buildImageInAzure = 'buildImageInAzure';
+const buildFromProjectLabels: string[] = [
+    localize('azure', 'Build from project remotely using Azure Container Registry')
+    //localize('docker', 'Build from project locally using Docker')
```
Comment (alexweininger) [inline src/commands/deploy/buildImageInAzure/BuildFromProjectListStep.ts:23]: I probably missed something, but what are we commenting this out until it's ready somewhere else?

Comment (motm32) [reply src/commands/deploy/buildImageInAzure/BuildFromProjectListStep.ts:23]: This is related to another create CA scenario (#285). This scenario won't be in this release so just putting a placeholder for now.

**Review by nturinski — APPROVED** [review]

**Pull request merged**